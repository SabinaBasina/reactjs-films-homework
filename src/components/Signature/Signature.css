.about {
  background-color: rgba(0, 0, 0, 0.753);
}

.text {
  color: white;
}
