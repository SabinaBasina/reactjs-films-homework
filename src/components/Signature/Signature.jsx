import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import styles from './Signature.scss';

class Signature extends PureComponent {
  render() {
    const { name } = this.props;
    return (
      <main>
        <h1>
          {`Hello, ${name}`}
        </h1>
        <article className={styles.about}>
          <p className={styles.text}>
            Text
          </p>
        </article>
      </main>
    );
  }
}

Signature.propTypes = {
  name: PropTypes.string,
};

Signature.defaultProps = {
  name: 'undefined',
};

export default Signature;
