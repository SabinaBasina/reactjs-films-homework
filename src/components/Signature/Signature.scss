$bg-color: rgb(252, 141, 159);

h1 {
  color: $bg-color;
}

.about {
  background-color: rgba(0, 0, 0, 0.753);
}

.text {
  color: white;
}
