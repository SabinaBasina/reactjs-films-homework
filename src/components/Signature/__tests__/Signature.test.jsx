import React from 'react';
// import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import Signature from '../Signature';

/* global test, expect */

test('Signature test', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Signature />);
  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
