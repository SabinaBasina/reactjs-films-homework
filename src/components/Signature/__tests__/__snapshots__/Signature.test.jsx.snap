// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signature test 1`] = `
<main>
  <h1>
    Hello,
    undefined
  </h1>
  <article
    className="about"
  >
    <p
      className="text"
    >
      Text1
    </p>
  </article>
</main>
`;
